test/org/sagebionetworks/bridge/dynamodb/DynamoFPHSExternalIdentifierDaoTest.java

```diff
@@ -0,0 +1,133 @@
+        
+        assertTrue(dao.verifyExternalId(new ExternalIdentifier("bar")));
+        assertFalse(dao.verifyExternalId(new ExternalIdentifier("baz")));
+    }
+    
+    @Test(expected = EntityNotFoundException.class)
+    public void registerUnknownCodeThrowsException() throws Exception {
+        dao.registerExternalId(new ExternalIdentifier("bar"));
+    }
+    
+    @Test(expected = EntityAlreadyExistsException.class)
+    public void registerCodeAgainThrowsException() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1));
+        dao.registerExternalId(new ExternalIdentifier(id1.getExternalId()));
+        
+        dao.registerExternalId(new ExternalIdentifier("baz"));
+    }
+    
+    @Test
+    public void registerValidCodeOK() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1));
+        
+        dao.registerExternalId(new ExternalIdentifier("baz"));
+        
+        assertTrue(dao.getExternalIds().get(0).getRegistered());
+    }
+    
+    @Test
+    public void getExternalIds() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("bar");
+        DynamoFPHSExternalIdentifier id2 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1, id2));
+        dao.registerExternalId(new ExternalIdentifier(id1.getExternalId()));
+        
+        List<FPHSExternalIdentifier> identifiers = dao.getExternalIds();
+        assertEquals(2, identifiers.size());
+
+        FPHSExternalIdentifier identifier = getById(identifiers, "bar");
+        assertEquals("bar", identifier.getExternalId());
+        assertTrue(identifier.getRegistered());
+
+        identifier = getById(identifiers, "baz");
+        assertEquals("baz", identifier.getExternalId());
+        assertFalse(identifier.getRegistered());
+    }
+
+    @Test
+    public void addExternalIds() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("bar");
+        DynamoFPHSExternalIdentifier id2 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1, id2));
+
+        // Register id1. It should stay registered.
+        dao.registerExternalId(new ExternalIdentifier(id1.getExternalId()));
+        
+        // Add a third ID, you will have three IDs, none are registered
+        id1 = new DynamoFPHSExternalIdentifier("bar");
+        id2 = new DynamoFPHSExternalIdentifier("baz");
+        DynamoFPHSExternalIdentifier id3 = new DynamoFPHSExternalIdentifier("boo");
+        dao.addExternalIds(Lists.newArrayList(id1, id2, id3));
+        
+        // Two updates by primary key, and an add for 3 records
+        List<FPHSExternalIdentifier> identifiers = dao.getExternalIds();
+        assertEquals(3, identifiers.size());
```

Comment:
Again changed (we already pull out specific results by ID to verify they are somewhere in the list)